src/error.rs

```diff
@@ -6,92 +6,117 @@ use std::sync::Arc;
 /// An enum containing all kinds of game framework errors.
 #[derive(Debug)]
 pub enum GameError {
```

Comment:
Using [`thiserror`](https://crates.io/crates/thiserror) would really do some justice to this code.

Reply:
Note I am not a maintainer, just thought I would drop my opinion

Reply:
I know it exists, but I'd rather not use it to improve compile times, as it only makes a difference on the internal cleanliness of the code and not outside.

Reply:
I agree that `thiserror` isn't really necessary here, though it might make things a bit simpler to write.

Reply:
Shouldn't we first use the `--timings` flag to check what is causing slow build times? Because I don't think `thiserror` will cause a large deviation.

Reply:
Honestly I just don't care about shedding that boilerplate code. It's not that much and it's perfectly readable. Sure using a macro like `thiserror` would make it look more tidy. But to me trading in compile time, even if just very very little of it, for that just isn't worth it, sorry.